src/Poker/Holdem/Simulate.hs

```diff
@@ -0,0 +1,155 @@
+-- |
+-- Module      : Poker.Holdem.Simulate
+-- Description : Monte Carlo simulation of a poker holdem game.
+-- Copyright   : (c) Ghais Issa, 2021
+--
+-- Uses Monte Carlo Methods to determine the probability for each player in a game winning.
+-- The simulation randomize each of the unknown cards in the game, this could include any
+-- number of players' cards as well as flop, turn and river.
+module Poker.Holdem.Simulate
+  (
+    Player(..)
+  , Game(..)
+  , simulate
+  ) where
+
+import           Data.Random (RVar, RandomSource, runRVar)
+
+import           Control.Monad.State
+import           Data.List (transpose)
+
+import           Data.Random.Source.DevRandom (DevRandom (DevRandom))
+import           Poker.Deck
+import qualified Poker.Deck as Deck (shuffle)
+import           Poker.Holdem
+import           Poker.Holdem.Evaluate (HandRank, evaluate)
+
+-- | A player can have 0, 1, 2 known cards.
+data Player = Player
```

Comment:
One of those very-specific-definition things, where many applications will know both cards. It'd be nice imo not to have this in the "poker" package